openstack_plugin/resources/volume/volume.py

```diff
@@ -498,9 +499,13 @@ def delete(openstack_resource):
     # otherwise we should check the if the volume is really deleted or not
     # by keep calling get volume api
     if VOLUME_TASK_DELETE not in ctx.instance.runtime_properties:
-        # Delete volume resource
-        openstack_resource.delete()
-        ctx.instance.runtime_properties[VOLUME_TASK_DELETE] = True
+        if ctx.instance.runtime_properties.get(RESOURCE_ID):
+            ctx.logger.info('Volume is already uninitialized.')
```

Comment:
@0lvin Same here, we can move this from here and just add the `ctx.instance.update()` if it is needed

Reply:
i will rework it, looks it has mistake. We need ignore delete if we don't have any `RESOURCE_ID`.